src/Auth/AuthServiceProvider.php

```diff
@@ -15,14 +15,21 @@ public function boot(Dispatcher $dispatcher): void
         $dispatcher->listen(RegisterDirectiveNamespaces::class, static fn (): string => __NAMESPACE__);
     }
 
-    public static function guard(): ?string
+    /**
+     * @return string[]
+     */
+    public static function guards(): array
     {
         $config = Container::getInstance()->make(ConfigRepository::class);
-        $lighthouseGuard = $config->get('lighthouse.guard');
         $guards = $config->get('auth.guards');
 
-        return isset($guards[$lighthouseGuard])
-            ? $lighthouseGuard
-            : $config->get('auth.defaults.guard');
+        $lighthouseGuards = array_filter(
+            (array) $config->get('lighthouse.guards', []),
+            static fn (string $guard): bool => isset($guards[$guard]),
+        );
```

Comment:
I think we should throw if there is a mismatch here

Reply:
Nice idea,

I am very bad at naming (and english too obviously) ^^ would this be ok ?
InvalidConfigurationException("Guard ${guard} is not a known guard.")

Reply:
This obviously become way more readable :) 

```php
public static function guards(): array
{
    $config = Container::getInstance()->make(ConfigRepository::class);
    
    return $config->get('lighthouse.guards', [$config->get('auth.guards')]);
}
```